LanguageFeatures/Augmentation-libraries/scoping_A01_t05.dart

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion If a member declaration needs to reference a static or instance
+/// member declared in another introductory or augmenting declaration of the
+/// same type, it can use `this.name` for instance members an `TypeName.name`
+/// for static members to be absolutely sure. Or it can rely on the default if
+/// `name` is not in the lexical scope at all, in which case it’s interpreted as
+/// `this.name` if it occurs inside a scope where a `this` is available.
+///
+/// @description Checks that `name` is interpreted as `this.name` if it occurs
+/// inside a scope where a `this` is available. Test a variable.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+import augment 'scoping_A01_t05_lib.dart';
+import '../../Utils/expect.dart';
+
+const foo = "Global foo";
+const bar = "Global bar";
+
+class C {
+  static const foo = "foo";
+  String bar = "bar";
+
+  void testAugmenting() {
+    Expect.equals("Global baz", baz);
+    Expect.equals("qux", qux);
```

Comment:
This one should actually yield "Global qux": The lexical lookup does not find any declarations whose name is `qux` in the current scope, the formal parameter scope of `testAugmenting`, or the body scope of this `class C` declaration, but it _does_ find a declaration named `qux` in the library scope (because the library scope contains bindings for all names of declarations at the library level, which includes top-level declarations of part files).

Agreed, it's surprising that the library scope contains everything from all files, but the class body scope only contains bindings for the declarations that occur in the same declaration (so we do not have lexical access to names that are declared in other declarations of the same enclosing entity, whether it's in the same file or in different files). Surprising, but currently true.

See also https://github.com/dart-lang/language/issues/4061. ;-)

Reply:
Still, we should treat this discussion as we've treated other discussions: Follow the spec for now, change tests if and when the spec is modified.

Reply:
Other cases are similar.